chart/prometheus-msteams/templates/configMapTemplate.yaml

```diff
@@ -5,8 +5,10 @@ metadata:
  name: {{ template "app.name" . }}-card-template
 binaryData:
   card.tmpl: |-
-{{- if .Values.customCardTemplate }}
-{{ .Values.customCardTemplate | b64enc | indent 4 }}
-{{- else }}
 {{ .Files.Get "card.tmpl" | b64enc | indent 4 }}
+{{- range $index, $customCardTemplate := .Values.connectorsWithCustomTemplates }}
+{{- if hasKey $customCardTemplate "template_file" }}
+  custom_card_{{ $index }}.tmpl: |-
+{{ $customCardTemplate.template_file | b64enc | indent 4 }}
+{{- end }}
```

Comment:
unfortunately this does not work when using `--set-file`. I was trying a little bit whether I can fix it, but don't have a solution yet. Somehow we need to do a dynamic `{{ .Files.Get <filename> | b64enc | indent 4 }}`.

Reply:
Hmm, works fine for me. This is how I executed it

```
helm template chart/prometheus-msteams/ -f values.yaml --set-file "connectorsWithCustomTemplates[0].template_file=custom-message-card.tmpl"
```

This is the content of `values.yaml`

```
connectorsWithCustomTemplates:
  - request_path: /alert2
    template_file:
    webhook_url: https://webhook
    escape_underscores: true
```

And this is the content of `custom-message-card.tmpl`

```
{{ define "teams.card" }}
{...}
{{ end }}
```

Output of the config map is

```
# Source: prometheus-msteams/templates/configMapTemplate.yaml
apiVersion: v1
kind: ConfigMap
metadata:
 name: prometheus-msteams-card-template
binaryData:
  card.tmpl: |-
    e3sgZGVmaW5lICJ0ZWFtcy5jYXJkIiB9fQp7CiAgIkB0eXBlIjogIk1lc3NhZ2VDYXJkIiwKICAiQGNvbnRleHQiOiAiaHR0cDovL3NjaGVtYS5vcmcvZXh0ZW5zaW9ucyIsCiAgInRoZW1lQ29sb3IiOiAie3stIGlmIGVxIC5TdGF0dXMgInJlc29sdmVkIiAtfX0yREM3MkQKICAgICAgICAgICAgICAgICB7ey0gZWxzZSBpZiBlcSAuU3RhdHVzICJmaXJpbmciIC19fQogICAgICAgICAgICAgICAgICAgIHt7LSBpZiBlcSAuQ29tbW9uTGFiZWxzLnNldmVyaXR5ICJjcml0aWNhbCIgLX19OEMxQTFBCiAgICAgICAgICAgICAgICAgICAge3stIGVsc2UgaWYgZXEgLkNvbW1vbkxhYmVscy5zZXZlcml0eSAid2FybmluZyIgLX19RkZBNTAwCiAgICAgICAgICAgICAgICAgICAge3stIGVsc2UgLX19ODA4MDgwe3stIGVuZCAtfX0KICAgICAgICAgICAgICAgICB7ey0gZWxzZSAtfX04MDgwODB7ey0gZW5kIC19fSIsCiAgInN1bW1hcnkiOiAie3stIGlmIGVxIC5Db21tb25Bbm5vdGF0aW9ucy5zdW1tYXJ5ICIiIC19fQogICAgICAgICAgICAgICAgICB7ey0gaWYgZXEgLkNvbW1vbkFubm90YXRpb25zLm1lc3NhZ2UgIiIgLX19CiAgICAgICAgICAgICAgICAgICAge3stIGlmIGVxIC5Db21tb25MYWJlbHMuYWxlcnRuYW1lICIiIC19fQogICAgICAgICAgICAgICAgICAgICAgUHJvbWV0aGV1cyBBbGVydAogICAgICAgICAgICAgICAgICAgIHt7LSBlbHNlIC19fQogICAgICAgICAgICAgICAgICAgICAge3stIC5Db21tb25MYWJlbHMuYWxlcnRuYW1lIC19fQogICAgICAgICAgICAgICAgICAgIHt7LSBlbmQgLX19CiAgICAgICAgICAgICAgICAgIHt7LSBlbHNlIC19fQogICAgICAgICAgICAgICAgICAgIHt7LSAuQ29tbW9uQW5ub3RhdGlvbnMubWVzc2FnZSAtfX0KICAgICAgICAgICAgICAgICAge3stIGVuZCAtfX0KICAgICAgICAgICAgICB7ey0gZWxzZSAtfX0KICAgICAgICAgICAgICAgICAge3stIC5Db21tb25Bbm5vdGF0aW9ucy5zdW1tYXJ5IC19fQogICAgICAgICAgICAgIHt7LSBlbmQgLX19IiwKICAidGl0bGUiOiAiUHJvbWV0aGV1cyBBbGVydCAoe3sgLlN0YXR1cyB8IHRpdGxlIH19KSIsCiAgInNlY3Rpb25zIjogWyB7eyRleHRlcm5hbFVybCA6PSAuRXh0ZXJuYWxVUkx9fQogIHt7LSByYW5nZSAkaW5kZXgsICRhbGVydCA6PSAuQWxlcnRzIH19e3stIGlmICRpbmRleCB9fSx7ey0gZW5kIH19CiAgICB7CiAgICAgICJhY3Rpdml0eVRpdGxlIjogIlt7eyAkYWxlcnQuQW5ub3RhdGlvbnMuZGVzY3JpcHRpb24gfX1dKHt7ICRleHRlcm5hbFVybCB9fSkiLAogICAgICAiZmFjdHMiOiBbCiAgICAgICAge3stIHJhbmdlICRrZXksICR2YWx1ZSA6PSAkYWxlcnQuQW5ub3RhdGlvbnMgfX0KICAgICAgICB7CiAgICAgICAgICB7ey0gaWYgbmUgJGtleSAiZGVzY3JpcHRpb24iIC19fQogICAgICAgICAgICAibmFtZSI6ICJ7eyAka2V5IH19IiwKICAgICAgICAgICAgInZhbHVlIjogInt7ICR2YWx1ZSB9fSIKICAgICAgICAgIHt7LSBlbmQgLX19CiAgICAgICAgfSwKICAgICAgICB7ey0gZW5kIC19fQogICAgICAgIHt7JGMgOj0gY291bnRlcn19e3sgcmFuZ2UgJGtleSwgJHZhbHVlIDo9ICRhbGVydC5MYWJlbHMgfX17e2lmIGNhbGwgJGN9fSx7eyBlbmQgfX0KICAgICAgICB7CiAgICAgICAgICAibmFtZSI6ICJ7eyAka2V5IH19IiwKICAgICAgICAgICJ2YWx1ZSI6ICJ7eyAkdmFsdWUgfX0iCiAgICAgICAgfQogICAgICAgIHt7LSBlbmQgfX0KICAgICAgXSwKICAgICAgIm1hcmtkb3duIjogdHJ1ZQogICAgfQogICAge3stIGVuZCB9fQogIF0KfQp7eyBlbmQgfX0K
  custom_card_0.tmpl: |-
    e3sgZGVmaW5lICJ0ZWFtcy5jYXJkIiB9fQp7Li4ufQp7eyBlbmQgfX0K
```

Reply:
late evening typo *facepalm*. Awesome, this works great! :). I really like that, thanks a lot @weimerb-tng!